test/WebJobs.Extensions.Tests/Extensions/Http/HttpTriggerBindingTests.cs

```diff
@@ -29,9 +29,12 @@ public class HttpTriggerBindingTests
             request.Content = new StringContent(input);
             var bindingData = await HttpTriggerAttributeBindingProvider.HttpTriggerBinding.GetRequestBindingDataAsync(request);
 
-            Assert.Equal(4, bindingData.Count);
+            Assert.Equal(5, bindingData.Count);
             Assert.Equal("testing", bindingData["test"]);
             Assert.Equal("123", bindingData["baz"]);
+
+            dynamic nestedObject = bindingData["nestedObject"];
```

Comment:
Or this would be fine too: 
   dynamic nestedObject = (JObject) bindingData["nestedObject"]; 
Think of the  extra (Jobj) cast as a micro-test that the return type is really JObj and doesn't get changed to IDict<string,obj> later on